htdocs/core/lib/price.lib.php

```diff
@@ -174,7 +174,7 @@ function calcul_price_total($qty, $pu, $remise_percent_ligne, $txtva, $uselocalt
 	// pu calculation from pu_devise if pu empty
 	if (empty($pu) && !empty($pu_devise)) {
 		if (!empty($multicurrency_tx)) {
-			$pu = $pu_devise / $multicurrency_tx;
+			$pu = price2num($pu_devise / $multicurrency_tx, 'MT');
```

Comment:
$pu is a unit price, also it is a foreign currency, so rounding must use 'UC' instead of 'MT'

Reply:
You are right, 'MT' is wrong. But you probably mean 'CU' instead of 'UC', because my system does not recognize 'UC'. 

I have tested it with 'CU'.
The problem with that is that then the value for “Total (excl. in currency)” is wrong by 0.01.

![2024-10-14_11-41-33](https://github.com/user-attachments/assets/f7c5c89d-3dfc-4cf4-bb82-932cc1e397c4)

Then I tested it with 'MU'.
Here the value for “Total (excl. in currency)” is calculated correctly. However, the value for “Remaining unpaid, original currency” is not correct because it is calculated separately from the original currency and the exchange rate.

![2024-10-14_11-46-02](https://github.com/user-attachments/assets/d985b50e-0322-4f2c-baf6-36a43d10c81f)

I don't know if and how this can be solved in a good way.

Reply:
I doubt this can always be fixed. Main currency and foreign currency are using different accuracy, rounding rules, so it is not possible to have always no rounding error in both main currency and foreign currency.